app/connectors/UpdateIncomeSourceConnector.scala

```diff
@@ -21,50 +21,53 @@ import models.updateIncomeSource.request.UpdateIncomeSourceRequestModel
 import models.updateIncomeSource.{UpdateIncomeSourceResponse, UpdateIncomeSourceResponseError, UpdateIncomeSourceResponseModel}
 import play.api.http.Status
 import play.api.http.Status.OK
-import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
+import play.api.libs.json.Json
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, StringContextOps}
 
 import javax.inject.Inject
-import scala.annotation.nowarn
 import scala.concurrent.{ExecutionContext, Future}
 
-//TODO: Remove suppression annotation after upgrading this file to use HttpClientV2
-@nowarn("cat=deprecation")
-class UpdateIncomeSourceConnector @Inject()(val http: HttpClient,
+class UpdateIncomeSourceConnector @Inject()(val http: HttpClientV2,
                                             val appConfig: MicroserviceAppConfig)
                                            (implicit ec: ExecutionContext) extends RawResponseReads {
 
   val updateIncomeSourceUrl: String = s"${appConfig.ifUrl}/income-tax/business-detail/income-source"
-  private val ifHeaders: Seq[(String, String)] = appConfig.getIFHeaders("1776")
+  private val headers: Seq[(String, String)] = appConfig.getIFHeaders("1776")
+
   def updateIncomeSource(body: UpdateIncomeSourceRequestModel)(implicit headerCarrier: HeaderCarrier): Future[UpdateIncomeSourceResponse] = {
     val url = updateIncomeSourceUrl
     logger.info("INFO " +
-      s"Calling PUT $url \n\nHeaders: $headerCarrier \nAuth Headers: $ifHeaders \nBody:$body")
+      s"Calling PUT $url \n\nHeaders: $headerCarrier \nAuth Headers: $headers \nBody:$body")
```

Comment:
I guess we can downgrade this logging to debug level, as this is not something we need in prod and other environments